### PR TITLE
feat: Added peers sorting on views

### LINKED ISSUE #296: [ Feature Request ] - v2
hello i'm using lont time wg-portal and thank u so much.
i'm asking Feature Request if... possile...

- Connected Current VPN Peers on the TOP
or singe page show only connected peers.

- Traffic .. now are in Bytes .....Traffic .. can we be in Mib ?

### PR DESCRIPTION
**Pull request by bonddim**
Resolves #296 

- Sorts connected peers first on **Interface** and **Profile** views by default.
- Added **RX/TX** column with sorting by total amount of bytes on peer.

image
image

Also enables sorting on table fields - **NAME**, **USER**,  **IP'S** , **STATUS**

image
image

### CODE REVIEW

File: frontend/src/helpers/utils.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function ipToLong(ip) {
+  return ip.split('.').reduce((acc, octet) => (acc << 8) + parseInt(octet, 10), 0);
```
Comment (h44z) [inline frontend/src/helpers/utils.js:2]: what about IPv6?

**Review by h44z — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/utils.js (hunk 1 above)
Comment (bonddim) [reply frontend/src/helpers/utils.js:2]: While IPv4 addresses are available by default, and both addresses are showed in one column, I think that sorting on IPv4 is enough.

Comment (h44z) [reply frontend/src/helpers/utils.js:2]: And what if a user only uses IPv6? parseInt might return NaN in that case

Comment (h44z) [reply frontend/src/helpers/utils.js:2]: maybe we can add a check for IPv6 (string contains ':'), just to make sure that the code always produces valid resutls

Comment (bonddim) [reply frontend/src/helpers/utils.js:2]: You're right, I missed this. Will update

**Comment by BliNeR-KeY** [comment]
looks really nice..
pls can add also on the CURRENT VPN PEERS also TRAFFIC..

NAME, USER, IP'S , STATUS  and TRAFFIC 

like wireguard app RX / TX.  i hope not in bytes .. 

it nned only float code for rx_bytes, tx_bytes

**Comment by bonddim** [comment]
> looks really nice.. pls can add also on the CURRENT VPN PEERS also TRAFFIC..
> 
> NAME, USER, IP'S , STATUS and TRAFFIC
> 
> like wireguard app RX / TX. i hope not in bytes ..
> 
> it nned only float code for rx_bytes, tx_bytes

will look into, but not in this PR

**Comment by BliNeR-KeY** [comment]
It works.
i buils standalone
Thank you a lot... is really important. 

image

**Comment by bonddim** [comment]
> It works.
> i buils standalone
> Thank you a lot... is really important. 
> 
> image
> 

Yep, and, as mentioned, you may sort on other columns also

File: frontend/src/helpers/utils.js (hunk 1 above)
Comment (bonddim) [reply frontend/src/helpers/utils.js]: Fixed
image

**Comment by bonddim** [comment]
> looks really nice.. pls can add also on the CURRENT VPN PEERS also TRAFFIC..
> 
> NAME, USER, IP'S , STATUS and TRAFFIC
> 
> like wireguard app RX / TX. i hope not in bytes ..
> 
> it nned only float code for rx_bytes, tx_bytes

Upd: Done

- Size will be adjusted automatically to "B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB".
- Sorting on this column will use total amount of RX/TX on peer.

image

**Comment by BliNeR-KeY** [comment]
hahha looks very nice tank youuuuuuu i will build soon.
thank you a lot.

i have more one request... 
but not too important... as isee module RX/TX already here.

wat you think add all WG interface RX/TX on the top ?

explain better
now RX/TX are only a peer..

wat you think add all WG i**nterface RX/TX** on the top ?

**Comment by bonddim** [comment]
Sounds good. 
@h44z , what do you think about adding extra property to interface?

**Comment by h44z** [comment]
> Sounds good. @h44z , what do you think about adding extra property to interface?

why not =) but lets make another PR for that feature

**Review by h44z — APPROVED** [review]

**Pull request merged**